### DISCUSSION TITLE
About direct conversion

### DISCUSSION THREAD

**Discussion by curtis18** [protomaps/PMTiles#148] [Q&A] [answered]
@bdon Thanks for your software. May I ask is there any plan to support converting GeoJSON to PMTiles directly? or is there any plan to integrate it with GDAL like MVT?

**Comment by bdon** [comment, marked as answer]
Direct GeoJSON to PMTiles is done via https://github.com/felt/tippecanoe . GDAL does have a basic "MVT" output but it attempts to act as a translator and is not a special-purpose generalization tool, so the results will always be worse. There are no particular plans to implement PMTiles in GDAL for this reason.

**Comment by curtis18** [comment]
Thanks for your clear explanation. May I ask also can PMTiles methodology work for 3D tiles like b3dm format? as it is a non single-file archive format as well.

**Comment by bdon** [comment]
I'm not familiar with b3dm. If it uses a pyramid addressing scheme like `0/0/0`, `1/0/0`, `1/0/1` etc then you can create a pmtiles archive with tile type `0` = other/unknown.

**Reply by bdon** [reply]
(moved to discussion)

**Reply by curtis18** [reply]
Understand and thanks.
